api/src/main/java/com/exelaration/abstractmemery/services/implementations/MemeServiceImpl.java

```diff
@@ -44,4 +45,21 @@ public String getMeme(int id) {
     }
     return memeData;
   }
+
+  public ArrayList<String> getMemes() {
+    ArrayList<String> memeNames = memeMetadataService.getMemes();
+    if (memeNames == null) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Images Do Not Exist");
```

Comment:
I'm still not sure if we should be actually throwing an error here. I believe the way this is set up, if a new user would go to the home page, they would receive a 404. I think we would just want it to be blank but thoughts?

Reply:
It shows up as a Meme Not Found in the frontend for the specific image. The rest of the images that do exist still show up

Reply:
Sorry I may have misinterpreted, let me check something

Reply:
![image](https://user-images.githubusercontent.com/42583816/80395503-4bacdc80-8881-11ea-956f-4c0a34308ce7.png)

Reply:
This is what shows up if there are no memes in the db

Reply:
I'm getting a 403

Reply:
that's because the JWTs are not fixed. The antmatchers are restricting you from accessing it because you are not authorized

Reply:
![image](https://user-images.githubusercontent.com/39968186/80395726-a2b2b180-8881-11ea-87ee-635843ebdccc.png)

Reply:
You have to change the antmatchers to "/**" in the websecurity file temporarily to see that it works

Reply:
Also there is a typo in the gallery_url. It should be "/meme" not "/meme/". I will fix that

Reply:
Yep, that works. Nice